### PR TITLE
Ab#115945 add lov to dropdowns

### PR DESCRIPTION
**Pull request by tropical-turtle**
Added Lov Service and improved performance. Fixed Localization of drop-downs and grid table.

### CODE REVIEW

File: src/CSF.SRDashboard.Client/Pages/SeafarerProfile.razor.cs [hunk 1]
```diff
@@ -101,15 +105,34 @@ public partial class SeafarerProfile
...
-        private int requestID;
+        [Inject]
+        public ILovService LovService { get; set; }
+
+        [Inject]
+        IHttpContextAccessor httpContextAccessor { get; set; }
+
+        public List<WorkloadRequestTableItem> TableItems;
+
+        List<SelectListItem> submissionMethods = new List<SelectListItem>();
```
[the remark this answers is not in the document]
Comment (tropical-turtle) [reply src/CSF.SRDashboard.Client/Pages/SeafarerProfile.razor.cs:124]: Changed

File: src/CSF.SRDashboard.Client/Services/Constants.cs [hunk 2]
```diff
@@ -47,14 +47,14 @@ public static class Constants
...
 
-        public static List<SelectListItem> SubmissionMethods = new List<SelectListItem> {
-            new SelectListItem { Id = "1", Text = "E-mail" },
-            new SelectListItem { Id = "2", Text = "Mail"},
-            new SelectListItem { Id = "3", Text = "Fax"},
-            new SelectListItem { Id = "4", Text = "Online"},
-            new SelectListItem { Id = "5", Text = "Phone"},
-            new SelectListItem { Id = "6", Text = "Other"}
-        };
+        //public static List<SelectListItem> SubmissionMethods = new List<SelectListItem> {
```
[the remark this answers is not in the document]
Comment (tropical-turtle) [reply src/CSF.SRDashboard.Client/Services/Constants.cs:50]: Removed

File: src/CSF.SRDashboard.Client/appsettings.ncd.json [hunk 3]
```diff
@@ -18,7 +18,8 @@
     "MarineMedicalGroupId": "fcc8a558-2e87-495b-9f38-718763983504"
   },
   "ApplicationInsights": {
-    "Instrumentationkey": "7b67ebb0-6876-4867-8359-1278cae44d3b"
+    //"Instrumentationkey": "7b67ebb0-6876-4867-8359-1278cae44d3b",
```
[the remark this answers is not in the document]
Comment (tropical-turtle) [reply src/CSF.SRDashboard.Client/appsettings.ncd.json:21]: Removed

File: src/CSF.SRDashboard.Client/appsettings.ncd.json [hunk 4]
```diff
@@ -33,10 +34,12 @@
...
     "GatewayToMpdis": "https://wwwappstestext.tc.gc.ca/Saf-Sec-Sur/4/csf-gmc-apigw-dev/mpdis",
     "WorkLoadManagement": "https://ncdsafsecsurapp-work-management-service.azurewebsites.net",
-    "LovService": "https://ncdsafsecsurapp-lov-service.azurewebsites.net",
-    "LovScopes": {
-      "Read": "https://034gc.onmicrosoft.com/ncd-lov-dev/ListOfValue.Read.All",
-      "CreateUpdate": "https://034gc.onmicrosoft.com/ncd-lov-dev/ListOfValue.CreateUpdate.All"
+
+    "LovServiceAPI": {
+      "BaseUrl": "https://tc-api-management-dev.azure-api.net/lov",
+      //"BaseUrl": "https://ncdsafsecsurapp-lov-service.azurewebsites.net",
```
[the remark this answers is not in the document]
Comment (tropical-turtle) [reply src/CSF.SRDashboard.Client/appsettings.ncd.json:40]: Removed

File: src/CSF.SRDashboard.Client/Pages/SeafarerProfile.razor.cs [hunk 5]
```diff
@@ -211,9 +237,119 @@ protected override async void OnAfterRender(bool firstRender)
...
+                        item.Status = statusAndPhase.Item1;
+                        item.ProcessingPhase = statusAndPhase.Item2;
+                    }
+                }
+            }
+
+            return tableItems;
+        }
+
+        private LovConstants.Languages GetCurrentLanguage()
```
[the remark this answers is not in the document]
Comment (tropical-turtle) [reply src/CSF.SRDashboard.Client/Pages/SeafarerProfile.razor.cs:288]: Moved that method into CommonService

File: src/CSF.SRDashboard.Client/Pages/ViewRequestDetails.razor.cs [hunk 6]
```diff
@@ -207,5 +216,50 @@ private List<StatusHistoryItem> PopulateStatusHistoryItems(List<WorkItemStatusDT
...
+            catch (FormatException)
+            {
+                Console.WriteLine("Stored in string");
+            }
+
+            return isStoredInInteger;
+
+        }
+
+        private LovConstants.Languages GetCurrentLanguage()
```
[the remark this answers is not in the document]
Comment (tropical-turtle) [reply src/CSF.SRDashboard.Client/Pages/ViewRequestDetails.razor.cs:248]: Moved that method into CommonService

**Review by TC-Cory — COMMENTED** [review]
Nice work! Could not detect issues with code. I sent a message to confirm one concern

**Review by TC-Cory — APPROVED** [review]

**Pull request merged**